### DISCUSSION TITLE
Exporting to PNG can take even up to 20 seconds

### DISCUSSION THREAD

**Discussion by mwanagosos** [clientIO/joint#1793] [Q&A] [answered]
Hi. I'm using the `paper.toPNG` function to export my graph to PNG. Unfortunately, it can take even up to 20 seconds. Is there any way to optimize it?

**Comment by kumilingus** [comment, marked as answer]
Use `useComputedStyles: false` option. It will skip the collection of external CSS styles. If you use external CSS styles, you have to provide them with `stylesheet` option. An example of this is to be found in the KitchenSink application.

**Reply by mwanagosos** [reply]
That did cut the time to less than a second, awesome. Unfortunately, even the `@clientio/rappid/rappid.css` is not included anymore and it breaks how the graph is displayed. I understand that I need to add it manually, right?

**Reply by mwanagosos** [reply]
Doing
```
import rappidStylesheet from '!raw-loader!@clientio/rappid/rappid.css';
```
and using it as the `stylesheet` option is a step in the right direction :+1:

**Reply by kumilingus** [reply]
Right. If I recall correctly, those are only CSS rules that affect the SVG content. (And if you care about performance I would strongly suggest not to use `scalable` group and legacy `joint.dia.Link`.

https://github.com/clientIO/joint/blob/master/css/layout.css#L101-L170
https://github.com/clientIO/joint/blob/master/css/layout.css#L63-L66
